### ISSUE TITLE
Docker for Mac Failing to Start

### ISSUE THREAD

**Issue by djbingham** [docker/for-mac#175]
### Expected behavior

On launch, Docker 1.12 for Mac is failing to start, with the status showing as "Docker starting..."

### Actual behavior

Docker for Mac should start and the status should show "Running".

### Information

Diagnostic ID: 73D8E9C0-4390-4395-A2AA-D315819B2EC8
Docker for Mac: 1.12.0 (Build 10871)
macOS: Version 10.11.5 (Build 15F34)
[ERROR]  docker-cli
         docker ps failed
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

I don't have reliable steps to reproduce as I have no idea what (if anything) I did that caused this. I was working fine with Docker for Mac, but suddenly all docker commands stopped working. I don't recall seeing any error messages, just commands such as `docker ps` did nothing. I think I cancelled a script that was executing various docker build/run commands shortly before docker stopped responding.

I then tried to restart Docker for Mac, which got stuck in the "Docker starting..." state. In an attempt to remove any potential conflicts, I have since followed the steps in https://github.com/docker/toolbox/blob/master/osx/uninstall.sh to remove my old Docker Toolbox installation, then restarted my Mac and I still have Docker for Mac stuck on "Docker starting..."

I'm happy to help out with any further tests on my system, but would prefer to avoid reinstalling Docker for Mac as it will take the best part of a day to re-download all of my images and longer to rebuild local containers.

**UPDATE:** I've just been reminded what the error message was that I saw before trying to restart Docker for Mac:

> Error response from daemon: Bad response from Docker engine.

I'm now getting the same error when I try running `docker ps` with Docker stuck in the "Docker starting..." state.

Also, probably worth mentioning that every time I restart Docker for Mac now, a system notification appears informing that it has just updated to version 1.12.

**Comment by djbingham** [comment]
I got impatient and reset to factory defaults. Docker seems to be working fine now, just going through all my projects downloading required images and building containers.

I do think this issue needs investigating even though a reset fixes it. Clearly something went wrong with Docker that caused it to get into a completely broken state, so without the cause being identified I wouldn't like to see this ticket closed.

**Comment by cmPrasanna** [comment]
+1: We are also facing the same issue, I am not sure what causes this. This definitely needs immediate attention.

**Comment by tony-garcia** [comment]
+1 ran into this today with Docker 1.12.1-rc1-beta23

**Comment by fjoesne** [comment]
+1 stuck on this. Version 1.12.0-a (build: 11213) tried factory reset.. no success

**Comment by levans002** [comment]
Resetting to factory defaults worked for me. +1 for a real

**Comment by cmPrasanna** [comment]
Moving to docker toolbox out of frustration. VirtualBox has been stable for me. Docker team please stop this movement out of this real nice option for mac users.

**Comment by codyseibert** [comment]
+1 Version 1.12.0-a (build: 11213)

**Comment by mchiang0610** [comment]
Merging with #119

**Comment by samoht** [comment]
Detected symptom of problem 'invariant-violated' in 73D8E9C0-4390-4395-A2AA-D315819B2EC8/20160808-154931.

An invariant within the code of one of the components was not held. Docker needs to be restarted.

May be related to docker/for-mac#89

The following log matches:
```
Aug  1 09:17:08 David-Hs-MacBook-Pro-2 Docker[com.docker.driver.amd64-linux][6682] <Notice>: VM: 2016-08-01 08:17:08 +0000 UTC kern.info kernel: docker0: port 7(veth39abd87) entered forwarding state
Aug  1 09:18:05 David-Hs-MacBook-Pro-2 Docker[com.docker.osxfs][6679] <Notice>: Volume.stop c5c04763e6aa3c6f48358faef36a836822cfedfee22ed82377dcb14b5f0b0f01 (paths = [/Users/david/sites/ft/version-service, /Users/david/sites/ft/version-service/linux_node_modules, /Users/david/.npm/])
Aug  1 09:18:05 David-Hs-MacBook-Pro-2 Docker[com.docker.osxfs][6679] <Error>: WATCH SUBSCRIBER REMOVAL INVARIANT VIOLATED
Aug  1 09:18:05 David-Hs-MacBook-Pro-2 Docker[com.docker.driver.amd64-linux][6682] <Notice>: VM: 2016-08-01 08:18:05 +0000 UTC daemon.info vsudd: 1284 Done. read: 265 written: 3847
Aug  1 09:18:05 David-Hs-MacBook-Pro-2 Docker[com.docker.driver.amd64-linux][6682] <Notice>: VM: 2016-08-01 08:18:05 +0000 UTC daemon.info vsudd: Connection 1286 to: 2376 from: 00000000
Aug  1 09:18:05 David-Hs-MacBook-Pro-2 Docker[com.docker.driver.amd64-linux][6682] <Notice>: VM: 2016-08-01 08:18:05 +0000 UTC kern.info kernel: vethc566098: renamed from eth0
```

**Comment by dsheets** [comment]
The invariant violation issue is not the root cause of this report. This particular invariant violation issue was fixed in Beta 23/Stable 1.12.1.
